index.js

```diff
@@ -403,6 +393,20 @@ module.exports = class NoiseSecretStream extends Duplex {
     this._rawStream.write(buf)
   }
 
+  _setupSecretBox (handshakeHash) {
+    this._boxSecret = b4a.allocUnsafeSlow(64) // encrypt<32>, decrypt<32>
+    this._boxSeq = b4a.allocUnsafeSlow(16) // increment<8>, initial<8>
+
+    const inputs = [b4a.concat([NS_INITIATOR, NS_BOX]), b4a.concat([NS_RESPONDER, NS_BOX])]
+    if (!this.isInitiator) inputs.reverse()
+
+    sodium.crypto_generichash(this._boxSecret.subarray(0, 32), inputs[0], handshakeHash)
```

Comment:
if you use `crypto_generichash_batch` here you don't have to concat the inputs, just pass the array directly

Reply:
oh nice, yes fixing